### ISSUE TITLE
[FR/Q] How can I delete all drawings from the current file/page?

### ISSUE THREAD

**Issue by NightMachinery** [ahrm/sioyek#928]
`delete_freehand_drawings` needs selecting a rectangle with the mouse, which is not easy to use for these purposes.

PS: Where are the drawings being saved? Do they work with `embed_annotations`? Can the color/size of the pointer be changed? Can a list of all drawings be seen ala `goto_highlight`?

**Comment by ahrm** [comment]
I don't think currently there is a way to do that, obviously the freehand drawing mode is targeted more towards mouse/touch users not keyboard users.

> Where are the drawings being saved? 

They are stored in a file next to the PDF file.

> Do they work with embed_annotations?

Yes.

>Can the color/size of the pointer be changed?

Yes. Using `set_freehand_type` and `set_freehand_thickness` commands.

> Can a list of all drawings be seen ala goto_highlight?

No, it is pretty easy to implement but I don't think it makes much sense. We can't really show a helpful searchable label for the drawings.

**Comment by NightMachinery** [comment]
@ahrm The delete commands are needed for mouse usage; I often want to draw some stuff during a presentation (sometimes just to direct attention to a particular location on the page), and then in the next presentation I want to delete all previous drawings. This is very difficult to do currently. 

While deleting the drawings on the current page is easier using the mouse, it's still not as easy as just pressing a key to reset the current page's drawings. Basically, the current delete command is great for undoing a mistake, but it's not suited for use during presentations.

**Comment by ahrm** [comment]
Added `clear_current_page_drawings` command  in https://github.com/ahrm/sioyek/commit/45256f51df6d5495077de8cbef773021b0783a76.

**Comment by NightMachinery** [comment]
> Added `clear_current_page_drawings` command in [45256f5](https://github.com/ahrm/sioyek/commit/45256f51df6d5495077de8cbef773021b0783a76).

Thanks. For deleting all the drawings of the current file, I should just delete the `.pdf.sioyek.drawings` file? Can I just do this?
```
new_command delete_current_file_drawings "%{file_path}sioyek.drawings"
```

BTW, storing the drawings in a file is great. Why not also do this for highlights? Plain text files near the original PDF files can be easily put under version control, shared, and parsed/manipulated by third-party tools. Using a centralized database makes no sense for per-file data IMO.

**Comment by NightMachinery** [comment]
```
new_command delete_current_file_drawings rm -f "%{file_path}sioyek.drawings"
```
This doesn't seem to work, the file gets recreated by sioyek, and sioyek doesn't update its internal state.

**Comment by ahrm** [comment]
Added `clear_current_document_drawings` command in https://github.com/ahrm/sioyek/commit/83974883cc6ce4bdb6c62fc6dcccbb3339f88ea7.

> BTW, storing the drawings in a file is great. Why not also do this for highlights? Plain text files near the original PDF files can be easily put under version control, shared, and parsed/manipulated by third-party tools. Using a centralized database makes no sense for per-file data IMO.

The main reason we store annotations in a single file is that they can be searched globally (this is not necessary for drawings because they can't be searched anyway). We do have a `write_annotations_file` command which creates an annotations file similar to the drawings file next to the pdf document.

**Comment by NightMachinery** [comment]
@ahrm Can I use `startup_commands` to set `set_freehand_type` and `set_freehand_thickness`?

**Comment by ahrm** [comment]
Yes:
```
startup_commands set_freehand_type(b);set_freehand_thickness(10)
```

**Comment by NightMachinery** [comment]
> Yes:
> 
> ```
> startup_commands set_freehand_type(b);set_freehand_thickness(10)
> ```

The size is set correctly, but the color is not. I also tried:
`startup_commands set_freehand_type("b")`, but that didn't work either.

**Comment by ahrm** [comment]
What is your exact `prefs.config`?

**Comment by NightMachinery** [comment]
> What is your exact `prefs.config`?

You are right, I had a typo there. It works correcrtly, sorry.

**Comment by NightMachinery** [comment]
@ahrm Does `clear_current_document_drawings` save the cleaned annotations? I find that after restarting my computer (it crashes every now and then), the deleted annotations are back.

**Comment by ahrm** [comment]
No, there is a bug where I forgot to add `is_drawing_dirty = true` after clearing the drawings.

**Comment by ahrm** [comment]
This is already fixed in the development branch btw.
